### PR TITLE
heuristics to help users who mixup legacy/new client API

### PR DESCRIPTION
**Pull request by pkasprzyk**
ref. https://neptune-labs.atlassian.net/browse/NPT-10404

### CODE REVIEW

File: neptune/__init__.py [hunk 1]
```diff
@@ -52,7 +53,19 @@
 ANONYMOUS_API_TOKEN = constants.ANONYMOUS_API_TOKEN
 
 
-def init(project_qualified_name=None, api_token=None, proxies=None, backend=None):
+CURRENT_KWARGS = ('project', 'run', 'custom_run_id', 'mode')
```
Comment (shnela) [inline neptune/__init__.py:56]: Shouldn't we have more of new-style `init` arguments?
* `custom_run_id`
* `mode`
* `name`
* `description`
* `tags`
* and so on...

File: neptune/new/run.py [hunk 2]
```diff
@@ -69,6 +71,19 @@ def inner_fun(self, *args, **kwargs):
     return inner_fun
 
 
+LEGACY_METHODS = (
+    'create_experiment',
```
Comment (shnela) [inline neptune/new/run.py:75]: `create_experiment`  isn't part of `Run` class.

It should be defined in `neptune.new` similar to other old `neptune` methods like:
* `set_project`
* `get_experiment`
* `append_tag`
* etc.

**Review by shnela — CHANGES_REQUESTED** [review]
I'd also add some tests.

File: neptune/__init__.py (hunk 1 above)
Comment (pkasprzyk) [reply neptune/__init__.py:56]: I was aiming for a "heuristic" approach: detect args that you can realistically expect to be used each time.

But maybe I'm too cunning for my own good. I'll add more args.

File: neptune/new/run.py (hunk 2 above)
Comment (pkasprzyk) [reply neptune/new/run.py:75]: 1. you're right, I'll add old `neptune` methods to `neptune.new`
2. I'd argue that this one actually makes sense, too. I think this is a legit flow for old API:
```
project = neptune.init(...)
project.create_experiment(...)
```

Comment (shnela) [reply neptune/new/run.py:75]: Ad 2.
Yes

**Review by shnela — DISMISSED** [review]

**Review by shnela — APPROVED** [review]

**Pull request merged**